File: pubsub/src/main/java/glitch/pubsub/events/CommerceEvent.java
package glitch.pubsub.events;

import glitch.api.ws.events.IEvent;
import glitch.pubsub.GlitchPubSub;
import glitch.pubsub.Topic;
import glitch.pubsub.object.json.CommerceMessage;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class CommerceEvent extends AbstractMessageEvent<CommerceMessage> implements IEvent<GlitchPubSub> {
    public CommerceEvent(GlitchPubSub client, Topic topic, CommerceMessage data) {
        super(client, topic, data);
    }
}

File: pubsub/src/main/kotlin/glitch/pubsub/events/CommerceEvent.kt
package glitch.pubsub.events

import glitch.api.ws.events.IEvent
import glitch.pubsub.GlitchPubSub
import glitch.pubsub.Topic
import glitch.pubsub.`object`.json.CommerceMessage

class CommerceEvent(client: GlitchPubSub, topic: Topic, data: CommerceMessage) : AbstractMessageEvent<CommerceMessage>(client, topic, data), IEvent<GlitchPubSub>
